src/libultra/voice/voicecontread2.c

```diff
@@ -1,3 +1,74 @@
-#include "global.h"
+/**
+ * File: voicecontread2.c
+ */
 
-#pragma GLOBAL_ASM("asm/non_matchings/boot/voicecontread2/__osVoiceContRead2.s")
+#include "ultra64/controller_voice.h"
+#include "io/controller.h"
+#include "functions.h"
+#include "variables.h"
+
+#define READ2FORMAT(p) ((__OSVoiceRead2Format*)(ptr))
```

Comment:
This one as well.

Reply:
How does this still work when `p` is different than `ptr`?

Reply:
Confused me as well, than I realized the local variable was called `ptr`. So the macro arg just gets dropped, but the `ptr` matches the local variable so it still works.